presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -460,6 +472,50 @@ private void checkTypesMatchForInsert(Insert insert, Scope queryScope, List<Colu
             }
         }
 
+        private void checkTypesMatchForNestedStructs(
+                Node node,
+                String errorMessage,
+                int i,
+                List<Type> columnRowTypes,
+                List<RowType.Field> expectedRowFields,
+                String fieldName)
+        {
+            if (columnRowTypes.size() != expectedRowFields.size()) {
+                throw new SemanticException(MISMATCHED_SET_COLUMN_TYPES,
+                        node,
+                        errorMessage + "Mismatch at column %d: '%s' has %d expression(s) but expected %d fields(s)",
+                        i + 1,
+                        fieldName,
+                        columnRowTypes.size(),
+                        expectedRowFields.size());
+            }
+            for (int j = 0; j < Math.max(columnRowTypes.size(), expectedRowFields.size()); j++) {
```

Comment:
If we reached here, then both the sizes are equal. We don't need Math.max.

Reply:
made changes